pkg/services/pluginsintegration/clientmiddleware/utils.go

```diff
@@ -23,3 +39,36 @@ type callResourceResponseSenderFunc func(res *backend.CallResourceResponse) erro
 func (fn callResourceResponseSenderFunc) Send(res *backend.CallResourceResponse) error {
 	return fn(res)
 }
+
+func requestStatusFromError(err error) (requestStatus, error) {
```

Comment:
Any reason why this returns an `error`, but it's always equal to the provided `err` (at least for now)?

Reply:
I think it was for convenience to use the return values from the function directly. I'll look into it